app.py

```diff
@@ -1 +1,521 @@
+                end = datetime.strptime(f'{date} {close}', datetime_format)
+
+                open_hours[date] = {
+                    'summary': None,
+                    'uid': None,
+                    'start': utils.to_utc_string(start),
+                    'end': utils.to_utc_string(end),
+                    'sequence': None,
+                    'recurrenceId': None,
+                    'lastModified': None
+                }
+            return open_hours
+
+    def generate_json_resources(self):
+        """
+        Generate resources and write to JSON files
+        """
+        base_url = self.config['locations_api']['url']
+        facil_locations = self.get_facil_locations()
+        gender_inclusive_restrooms = self.get_gender_inclusive_restrooms()
+        arcGIS_geometries = self.get_arcGIS_geometries()
+        locations = []
+
+        # Merge facil locations, gender inclusive restrooms and geometry data
+        for location_id, raw_facil in facil_locations.items():
+            raw_gir = gender_inclusive_restrooms.get(location_id)
+            raw_geo = arcGIS_geometries.get(location_id)
+            facil_location = FacilLocation(
+                raw_facil, raw_gir, raw_geo, self.proj
+            )
+
+            locations.append(facil_location)
+
+        # Send async calls and collect results
+        concurrent_calls = asyncio.gather(
+            self.get_dining_locations(),
+            self.get_extra_calendars()
+        )
+        loop = asyncio.get_event_loop()
+        concurrent_res = loop.run_until_complete(concurrent_calls)
+        loop.close()
+
+        # Concatenate locations
+        locations += self.get_extra_locations()  # extra locations
+        locations += self.get_extension_locations()  # extension locations
+        locations += self.get_parking_locations()  # parking locations
+        locations += concurrent_res[0]  # dining locations
+        locations += concurrent_res[1]['locations']  # extra service locations
+
+        extra_services = concurrent_res[1]['services']
+        campus_map_data = self.get_campus_map_data()
+        combined_locations = []
+        merge_data = []
+        for location in locations:
+            resource_id = location.calculate_hash_id()
+
+            # Merge with campus map data
+            if resource_id in campus_map_data:
+                campus_recourse = campus_map_data[resource_id]
+                location.address = campus_recourse['address']
+                location.description = campus_recourse['description']
+                location.descriptionHtml = campus_recourse['descriptionHTML']
+                location.images = campus_recourse['images']
+                location.thumbnails = campus_recourse['thumbnail']
+                location.website = campus_recourse['mapUrl']
+                location.synonyms = campus_recourse['synonyms']
+
+                if location.bldg_id == '0036':
```

Comment:
Could you add some documentation or a variable name about this building ID being the library?

Reply:
Sure, will do.